broadinstitute/single_cell_portal_core
# Resize legend upon resizing scatter plot (SCP-3795, SCP-3807)

This resizes the legend upon resizing the scatter plot.   Previously (#1211), the legend width and height did not update upon searching a gene, nor upon manually resizing the window.  Now it resizes to better fit updated dimensions.

Here's how it looks:

https://user-images.githubusercontent.com/1334561/140761314-1f3bd940-ebcc-4304-a39b-a4ec4476cae9.mov

An automated test was also added.  Tests involving window size seem very complicated, so I skip that here.  Instead, this new test adds automatic verification of the legend itself, as well as a basic legend interaction.  

The "mouse salmonella" synthetic study has also been enriched to give an easily reproducible test case for a cluster group with many labels, include long labels and labels with unintuitive formatting.  (This will be more relevant for my next PR to handle long labels, but seemed unproblematic to include here.)

To test:
* Go to a study with > 2 labels in a group-based annotation. Labels should ideally be short.
* Resize window height
* Verify legend resizes appropriately
* Resize window width
* Verify legend resizes appropriately
* Search a gene
* Verify legend resizes appropriately

This satisfies SCP-3795 and SCP-3807.
